programs/voter-stake-registry/src/instructions/close_voter.rs

```diff
@@ -13,17 +16,48 @@ pub struct CloseVoter<'info> {
         has_one = voter_authority,
         close = sol_destination)]
     pub voter: AccountLoader<'info, Voter>,
+
+    #[account(mut)]
     pub voter_authority: Signer<'info>,
+
+    #[account(mut)]
     pub sol_destination: UncheckedAccount<'info>,
+
+    pub token_program: Program<'info, Token>,
 }
 
-/// Closes the voter account, allowing one to retrieve rent exemption SOL.
+/// Closes the voter account (Optionally, also token vaults, as part of remaining_accounts),
+/// allowing one to retrieve rent exemption SOL.
 /// Only accounts with no remaining deposits can be closed.
-pub fn close_voter(ctx: Context<CloseVoter>) -> Result<()> {
+pub fn close_voter<'key, 'accounts, 'remaining, 'info>(
+    ctx: Context<'key, 'accounts, 'remaining, 'info, CloseVoter<'info>>,
+) -> Result<()> {
     let voter = &ctx.accounts.voter.load()?;
     let amount = voter.deposits.iter().fold(0u64, |sum, d| {
         sum.checked_add(d.amount_deposited_native).unwrap()
     });
     require!(amount == 0, VotingTokenNonZero);
+
+    let voter_seeds = voter_seeds!(voter);
+    for account in &mut ctx.remaining_accounts.iter() {
+        let token = Account::<TokenAccount>::try_from(&account.clone()).unwrap();
+        require!(token.owner == ctx.accounts.voter.key(), InvalidAuthority);
+        require!(token.amount == 0, VaultTokenNonZero);
```

Comment:
this is ok for now I think, but anyone could block a token account from being closable by sending 1 native token to it -- there'd be no way for the user to remove these funds and close the account.

We could keep the accounts frozen to avoid it. But that's an awkward complication. We could pass sol_destination's associated token accounts here, but that's also complex.

Keep as is, I think.